### DISCUSSION TITLE
Question about the large size of DLL compiled on Windows compared to Linux

### DISCUSSION THREAD

**Discussion by maniok098** [coin-or/Ipopt#795] [General]
Dear all,

first I really appreciate IPOPT, this wonderful tool and the fruitful discussions here in the forum :-)

I have been working with IPOPT some time on Windows, the compiled ipopt-3.dll, as provided under release page, is about 110MB,  and today I also tried IPOPT on my Arch Linux latop, the installed package size there was only 3.7MB, 

So why is the compiled code on Windows (the .dll data) so much larger than on Linux (I would assume that was a .so)?

**Comment by svigerske** [comment]
It's probably the included Intel MKL libraries that increase the size a lot. The ArchLinux package depends on Blas/Lapack libs to be installed.

**Reply by maniok098** [reply]
ahh, that's the point, thank you!

**Reply by svigerske** [reply]
> It's probably the included Intel MKL libraries that increase the size a lot.

The amount of MKL that ends up in the DLL also depends on how much of MKL is used. If Ipopt uses only some Blas/Lapack, the Ipopt DLL I get has only 14MB. If Ipopt includes the interface to Pardiso from MKL, then this goes up to 85MB.
